src/plugins/intel_gpu/include/intel_gpu/plugin/common_utils.hpp

```diff
@@ -111,6 +112,26 @@ inline ov::Shape predict_shape(const std::string& name, const cldnn::layout layo
     return layout.get_shape();
 }
 
+inline cldnn::memory::ptr allocate_memory_evenif_zero_bytes(cldnn::engine& _engine,
```

Comment:
Why did you move this logic out of engine::allocate_memory() ? My previous point was about not calling allocate_memory for 0 bytes at all if possible (i.e. if the buffer is not used). But in case if you actually need to save a scalar in memory, then I think we can have such logic within common allocate_memory()

Reply:
Zero bytes memory seems to be not common case but Loop op requires it where handles input/output/backedge memory.
I agreed original allocate_memory() function should not modify for some specific case, so I decided it need to be handled as util function to reinterpret. The util function could be moved to ocl_engine with current function name. Better idea?